code/modules/artifice/deadringer.dm

```diff
@@ -0,0 +1,183 @@
+/obj/item/weapon/deadringer
+	name = "silver pocket watch"
+	desc = "A fancy silver-plated digital pocket watch. Looks expensive."
+	icon = 'icons/obj/deadringer.dmi'
+	icon_state = "deadringer"
+	w_class = ITEMSIZE_SMALL
+	slot_flags = SLOT_ID | SLOT_BELT | SLOT_TIE
+	origin_tech = list(TECH_ILLEGAL = 3)
+	var/activated = 0
+	var/timer = 0
+	var/bruteloss_prev = 999999
+	var/fireloss_prev = 999999
+	var/mob/living/carbon/human/corpse = null
+	var/mob/living/carbon/human/watchowner = null
+
+
+/obj/item/weapon/deadringer/New()
+	..()
+	processing_objects |= src
+
+/obj/item/weapon/deadringer/Destroy() //just in case some smartass tries to stay invisible by destroying the watch
+	uncloak()
+	..()
+
+
+/obj/item/weapon/deadringer/dropped()
+	if(timer > 20)
```

Comment:
Why is this check here? If you drop the source of your cloak, you should probably automatically uncloak.

Reply:
It may be silly, but I use timer this way:
When it's above 20, it keeps its owner invisible, when it less or equal 20, it just goes into cooldown. Decided to conserve resources by using just one timer. If you feel there's a better solution to this, please tell.